listenbrainz/webserver/static/js/profile.js

```diff
@@ -0,0 +1,687 @@
+				console.log("Detected Spotify end of track, playing next track")
+				this.playNextTrack();
+				return;
+			}
+			this.setState({
+				currentSpotifyTrack: current_track,
+				playerPaused: paused,
+				errorMessage: null
+			});
+	}
+		
+	getAlbumArt(){
+		const track = this.state.currentSpotifyTrack;
+		if (!track || !track.album || !Array.isArray(track.album.images)){
+			return null
+		}
+		const sortedImages = track.album.images.sort((a,b)=>a.height > b.height ? -1 :1);
+		return sortedImages[0] &&
+		<div className="img-responsive" className="thumbnail"><img src={sortedImages[0].url} style={{width: "100%"}}/></div>;
+	}
+	
+	render(){
+		const playerButtonStyle = {width: '24%'};
+		return (
+			<div>
+				<div className="btn-group" role="group" aria-label="Playback control" style={{witdh: '100%'}}>
+
+					<button className="btn btn-default"
+						onClick={this.playPreviousTrack}
+						style={playerButtonStyle}>
+						<span className="fa fa-backward"></span> Prev
+					</button>
+
+					<button className="btn btn-default"
+						onClick={this.togglePlay}
+						style={playerButtonStyle}>
+						<span className={`${this.state.playerPaused ? 'hidden' : ''}`}>
+							<span className="fa fa-pause"></span>
+						</span>
+						<span className={`${!this.state.playerPaused ? 'hidden' : ''}`}>
+							<span className="fa fa-play"></span>
+						</span>
+						&nbsp;&nbsp;
+						{!this.state.playerPaused ? 'Pause' : 'Play'}
+					</button>
+
+					<button className="btn btn-default"
+						onClick={this.toggleDirection}
+						style={playerButtonStyle}>
+							{this.state.direction}
+							&nbsp;&nbsp;
+							<span className={`${this.state.direction === 'up' ? 'hidden' : ''}`}>
+								<span className="fa fa-angle-double-down"></span>
+							</span>
+							<span className={`${this.state.direction === 'down' ? 'hidden' : ''}`}>
+								<span className="fa fa-angle-double-up"></span>
+							</span>
+					</button>
+
+					<button className="btn btn-default"
+						onClick={this.playNextTrack}
+						style={playerButtonStyle}>
+						Next <span className="fa fa-forward"></span>
+					</button>
+
+				</div>
+				
+				{this.state.errorMessage && 
+					<div className="alert alert-danger" role="alert">
+						{this.state.errorMessage}
+					</div>
+				}
+				{this.state.currentSpotifyTrack && 
+					<div>
+						<h4>Currently playing:</h4>
+						<h3 style={{borderBottom: "1px solid lightgrey"}}>
+							{this.state.currentSpotifyTrack.name}
+							{this.state.currentSpotifyTrack.duration_ms &&
+								<span className="small"> — {millisecondsToHumanReadable(this.state.currentSpotifyTrack.duration_ms)}</span>
+							}
+						</h3>
+						<p>{this.state.currentSpotifyTrack.artists.map(artist => artist.name).join(', ')}</p>
+						<div>{this.getAlbumArt()}</div>
+					</div>
+				}
+				{this.props.currentListen && this.props.currentListen.user_name &&
+					<div>from {this.props.currentListen.user_name}'s listens</div>
```

Comment:
is the red here github's fault? (again, no big issue, just wondering)

Reply:
No clue.